modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/hooks/useEventListener.ts

```diff
@@ -24,7 +24,12 @@ import {useEffect} from 'react';
  * outside of your component's DOM (eg. attaching a "scroll" or "resize"
  * listener to the `window`).
  */
-export default function useEventListener(eventName, handler, phase, target) {
+export default function useEventListener(
+	eventName: string,
+	handler: EventListenerOrEventListenerObject,
```

Comment:
I didn't know these type of union types were directly available... need get on the TS wagon... do we have some training/resources for newcomers? Asking for a friend... 🙄

Reply:
There are zillions of resources out there but we should compile a list. The [official docs](https://www.typescriptlang.org/docs/) are great and there are good third-party resources like the [TypeScript Deep Dive](https://basarat.gitbook.io/typescript/).

In this specific instance though, my process is:

1. Want to figure out type for event listener.
2. Go to [`lib.dom.d.ts`](https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts) and find [something relevant](https://github.com/microsoft/TypeScript/blob/3b06ef1a733904013d67cb1e1fdb366082f0f9dd/lib/lib.dom.d.ts#L4941-L4942).

Note that I didn't literally go to the `lib.dom.d.ts` file on GitHub. I actually just used my "go to definition" keybinding inside Vim on `document.addEventListener` instance, and it jumped straight there (the `lib` files are all inside `node_modules`; they come with TypeScript). So this is one of those things that you don't know about until you know about it, but once you do, it is a treasure chest of useful info.

Reply:
> There are zillions of resources out there but we should compile a list.

[Made an issue for this](https://github.com/liferay/liferay-frontend-projects/issues/481).

Reply:
> So this is one of those things that you don't know about until you know about it, but once you do, it is a treasure chest of useful info

Might be good to share this invaluable information with our unsuspecting future devs ;)